lib/remap-functions/src/parse_grok.rs

```diff
@@ -103,15 +103,15 @@ impl Expression for ParseGrokFn {
 
                 Ok(Value::from(result))
             }
-            None => Ok(Value::from(BTreeMap::new())),
+            None => Err("unable to parse input with grok pattern".into()),
         }
     }
 
     fn type_def(&self, state: &state::Compiler) -> TypeDef {
         self.value
             .type_def(state)
-            .fallible_unless(value::Kind::Bytes)
-            .with_constraint(value::Kind::Array)
+            .into_fallible(true)
+            .with_constraint(value::Kind::Map)
```

Comment:
I don't fully understand how the type changed from `Array` to `Map`, could you help me out here?

Reply:
It didn't. It should never have been `Array` in the first place. I'm not sure what I was thinking at the time..

Until recently these types weren't really used much - I believe the only place was the condition system would check that the return was a Boolean. So the error was never spotted until I was looking over this file for this PR.

Recent changes mean that we do actually check the types much more, so if function expecting a Map as a parameter took the output from this function it would have incorrectly errored.

Reply:
I see - thanks!